compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -238,6 +238,7 @@ impl<'tcx> LayoutCalculator for LayoutCx<'tcx, TyCtxt<'tcx>> {
 pub enum SizeSkeleton<'tcx> {
     /// Any statically computable Layout.
     Known(Size),
+    Generic(ty::Const<'tcx>),
```

Comment:
This should definitely have its own separate documentation (including the fact that the constant has to be of type `usize` and represents the size of the type in bytes).